### PR TITLE
Overwriting properties in the read_materials_process

### LINKED ISSUE #1556: Overwriting properties in the read_materials_process
I encountered some problems with the read_materials_process, because i still had some properties (non-empty) defined in my old .mdpa file.

I was overwriting property 1 (that was also defined in the mdpa), and forgot to define a necessary value (```THICKNESS```). Now as the ```THICKNESS``` was defined in the old mdpa, it was just copied, without me noticing it.

I was wondering if it would make sense that the read_materials_process throws a warning if it tries to overwrite a property that already has values defined.

A function at the property.h would be needed to do so:
```c++
    bool IsEmpty()
    {
        return (mData.IsEmpty() or mTable.IsEmpty());
    }
```

This one could be called in the read_materials_process.

Whats your opinion?

### PR DESCRIPTION
**Pull request by armingeiser**
Add HasTables, HasVariables to properties and print an info message if the read_materials_process is modifiying them.

closes #1556

### CODE REVIEW

**Comment by philbucher** [comment]
Could you also implement the `IsEmpty` Method for the `Properties`?
~~~c
bool const IsEmpty()
{
        return !( HasVariables() || HasTable() );
}
~~~

**Comment by philbucher** [comment]
Thx!
Looks good to me, lets have @pooyan-dadvand have a look

File: kratos/includes/properties.h [hunk 1]
```diff
@@ -254,6 +259,16 @@ class Properties : public IndexedObject
 		mTables[Key(XVariable.Key(), YVariable.Key())] = rThisTable;
     }
 
+    bool const HasTables()
+    {
+        return (mTables.size() > 0);
```
Comment (pooyan-dadvand) [inline kratos/includes/properties.h]: You may use `mTables.empty()` as a slightly better implementation. (in some container it is faster)

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
They look ok to me. Only a minor comment.

**Comment by philbucher** [comment]
@pooyan-dadvand done

**Comment by armingeiser** [comment]
@philbucher thanks, did not find the time today

**Comment by pooyan-dadvand** [comment]
The change looks good to me, but seems that appveyor complaining...

**Comment by philbucher** [comment]
It was out of time (`Build execution time has reached the maximum allowed time for your plan (60 minutes).`)
and I dont know how to restart it

**Comment by philbucher** [comment]
@roigcarlo can we restart the appveyor manually?
I cannot find the option, even after logging in

**Comment by roigcarlo** [comment]
It should be possible if you login with the github account, but it works a bit oddly. For some reason the only account that can trigger restarts is mine so we made all members of the KratosGroup memebers of a Team in that account. It should appear a new login prompt if you try to loggin which lets you select "Kratos" which show allow you to restart the build.

Anyway I restarted it myself this time, I will keep investigating if there is any other easier solution.

**Comment by philbucher** [comment]
ok thx for the explanation

**Comment by philbucher** [comment]
@pooyan-dadvand now it compiled, all three checks are passing

**Comment by philbucher** [comment]
ping @pooyan-dadvand

**Review by pooyan-dadvand — APPROVED** [review]
:+1:

**Pull request merged**